echopype/echodata/echodata.py

```diff
@@ -315,114 +314,167 @@ def update_platform(
                 {trajectory_var: extra_platform_data[trajectory_var][0]}
             )
             extra_platform_data = extra_platform_data.drop_vars(trajectory_var)
-            extra_platform_data = extra_platform_data.swap_dims({"obs": time_dim})
+            obs_dim = list(extra_platform_data[time_dim].dims)[0]
+            extra_platform_data = extra_platform_data.swap_dims({obs_dim: time_dim})
+
+        def _extvar_properties(ds, name):
+            """Test the external variable for presence and all-nan values,
+            and extract its time dimension name.
+            Returns <presence>, <valid values>, <time dim name>
+            """
+            if name in ds:
+                # Assumes the only dimension in the variable is a time dimension
+                time_dim_name = ds[name].dims[0] if len(ds[name].dims) > 0 else "scalar"
+                if not ds[name].isnull().all():
+                    return True, True, time_dim_name
+                else:
+                    return True, False, time_dim_name
+            else:
+                return False, False, None
 
         # clip incoming time to 1 less than min of EchoData["Sonar/Beam_group1"]["ping_time"] and
         #   1 greater than max of EchoData["Sonar/Beam_group1"]["ping_time"]
         # account for unsorted external time by checking whether each time value is between
         #   min and max ping_time instead of finding the 2 external times corresponding to the
         #   min and max ping_time and taking all the times between those indices
-        sorted_external_time = extra_platform_data[time_dim].data
-        sorted_external_time.sort()
-        # fmt: off
-        min_index = max(
-            np.searchsorted(
-                sorted_external_time, self["Sonar/Beam_group1"]["ping_time"].min(), side="left"
-            ) - 1,
-            0,
-        )
-        # fmt: on
-        max_index = min(
-            np.searchsorted(
-                sorted_external_time,
-                self["Sonar/Beam_group1"]["ping_time"].max(),
-                side="right",
-            ),
-            len(sorted_external_time) - 1,
-        )
-        extra_platform_data = extra_platform_data.sel(
-            {
-                time_dim: np.logical_and(
-                    sorted_external_time[min_index] <= extra_platform_data[time_dim],
-                    extra_platform_data[time_dim] <= sorted_external_time[max_index],
-                )
-            }
-        )
+        def _clip_by_time_dim(external_ds, ext_time_dim_name):
+            # external_ds is extra_platform_data[vars-list-on-time_dim_name]
+            sorted_external_time = external_ds[ext_time_dim_name].data
+            sorted_external_time.sort()
+            # fmt: off
+            min_index = max(
+                np.searchsorted(
+                    sorted_external_time, self["Sonar/Beam_group1"]["ping_time"].min(), side="left"
+                ) - 1,
+                0,
+            )
```

Comment:
Just wanted to make it the same format as in max_index below
```suggestion
            min_index = max(
                np.searchsorted(
                    sorted_external_time,
                    self["Sonar/Beam_group1"]["ping_time"].min(),
                    side="left"
                ) - 1,
                0,
            )
```

Reply:
I like that. Thanks. I'll merge the commit.